homeassistant/components/arlo.py

```diff
@@ -38,14 +48,24 @@ def setup(hass, config):
     conf = config[DOMAIN]
     username = conf.get(CONF_USERNAME)
     password = conf.get(CONF_PASSWORD)
+    scan_interval = conf.get(CONF_SCAN_INTERVAL)
 
     try:
         from pyarlo import PyArlo
 
         arlo = PyArlo(username, password, preload=False)
         if not arlo.is_connected:
             return False
-        hass.data[DATA_ARLO] = arlo
+
+        # assign refresh period to base station thread
+        try:
+            if arlo.base_stations:
+                arlo_base_station = arlo.base_stations[0]
+                arlo_base_station.refresh_rate = scan_interval.total_seconds()
+        except (AttributeError, IndexError):
```

Comment:
Why would there be an attribute error?

The index error is solved by using my suggestion above.

Reply:
Yes, that is not needed anymore. Good refactoring!! Thanks